packages/prisma/schema.prisma

```diff
@@ -2203,3 +2207,34 @@ model WorkflowOptOutContact {
 
   @@unique([type, value])
 }
+
+model Role {
+  id          String           @id @default(cuid())
+  name        String
+  description String?
+  teamId      Int? // null for global roles
+  team        Team?            @relation(fields: [teamId], references: [id], onDelete: Cascade)
+  permissions RolePermission[]
+  memberships Membership[]
+  createdAt   DateTime         @default(now())
+  updatedAt   DateTime         @updatedAt
+  isGlobal    Boolean          @default(false)
+  isDefault   Boolean          @default(false) // Indicates if this is one of the default roles (OWNER, ADMIN, MEMBER)
```

Comment:
I like the addition of the global rules !

If `teamId = null` means global, then isn't `isGlobal` redundant? I think there's a chance we make a mistake in the future by putting teamId along with isGlobal=true.

Also `isDefault` isn't intuitive to me... what if we have `type` being `'system' | 'custom'`, or something like that? 🤔